### PR TITLE
add coredns dashboard

### LINKED ISSUE #7826: Dashboard and Alerts for CoreDNS
## Is your feature request related to a problem? Please describe

We lack the ability to get insight to CoreDNS easily over time.
Also, CoreDNS has metrics that are visible now in Grafana, but we lack triggers to create alerts and warnings.

## Describe the behaviour you'd like

We currently use image `rancher/mirrored-coredns-coredns:1.8.4`.
Use a recent version of a community dashboard to gain insight to CoreDNS.
Options [[1](https://grafana.com/grafana/dashboards/12539)][[2](https://grafana.com/grafana/dashboards/7279)][[3](https://grafana.com/grafana/dashboards/5926)], ranked in order of most recent, to oldest.

## Describe alternatives you've considered

[Other options here.](https://grafana.com/grafana/dashboards/?search=CoreDNS)

## Additional context

This video shows how to take an community dashboard and Gitpodify it. Thanks, @ArthurSens !

https://www.loom.com/share/ca15ea6ab12c465f9a72ff75e8f80ad6

In addition to customizing a community dashboard, you'll also need to simulate DNS failures, to assert that the alerts and warnings you build are triggered accordingly.

### PR DESCRIPTION
**Pull request by sagor999**
## Related Issue(s)

Fixes #7826 

## Release Notes

```release-note
none
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#7826*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sagor999** [comment]
@ArthurSens I tried to add coredns mixin, by building it and then just adding dashboard json file into our repo, but it fails without clear error as to why. I also feel like maybe there is a better way to import existing mixin from a git repo? Though I would like to import only dashboard and not alerts (if possible). 
Can you let me know what I am doing wrong here?

**Comment by ArthurSens** [comment]
Ah, the failure is that we have some automation that looks for the Dashboard data scheme and enforces it to follow some rules.

Unfortunately, Grafana's has been changing this data scheme left and right on every new version. I believe we're better off removing this enforcement and waiting for Grafana to agree on a stable scheme 😕.

I'll open a PR removing this from CI

**Comment by ArthurSens** [comment]
For importing mixins that are hosted in a Git repository, yes there are better ways 😅.

We use jsonnet-bundler to manage jsonnet dependencies. Notice that we have a file called [jsonnetfile.json](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/jsonnetfile.json), which is where we declare our dependencies. If you add coredns-mixin to this file, you can run `make vendor` inside that same directory and the dependency will be downloaded.

I've recorded myself doing some experiments with coredns-mixin and jsonnet here:
https://www.loom.com/share/858cafb2acfa4a0cb5fedad4b4d75847

**Comment by sagor999** [comment]
@ArthurSens I have been trying to test changes by deploying monitoring (to have local grafana with my changes) but the job keeps failing. Any idea what I might be doing wrong and how to fix this?

File: operations/observability/mixins/jsonnetfile.json
Comment (ArthurSens) [inline operations/observability/mixins/jsonnetfile.json:27]:
```suggestion
          "subdir": ""
```

The subdir must be the directory where `jsonnetfile.json` is located in that repo. For coredns-mixin it is the root dir :)

**Comment by ArthurSens** [comment]
> @ArthurSens I have been trying to test changes by deploying monitoring (to have local grafana with my changes) but the job keeps failing. Any idea what I might be doing wrong and how to fix this?

Just tried the new Makefile step that renders dashboards and it failed
image


Besides the fix in the dependencies file, I believe you forgot to add an import of that dependency. Here is the git diff that worked for me:

```diff
diff --git a/operations/observability/mixins/cross-teams/dashboards.libsonnet b/operations/observability/mixins/cross-teams/dashboards.libsonnet
index 4d207da4..c1ca1945 100644
--- a/operations/observability/mixins/cross-teams/dashboards.libsonnet
+++ b/operations/observability/mixins/cross-teams/dashboards.libsonnet
@@ -3,6 +3,8 @@
  * Licensed under the MIT License. See License-MIT.txt in the project root for license information.
  */
 
+local corednsMixin = (import 'github.com/povilasv/coredns-mixin/mixin.libsonnet');
+
 (import './dashboards/gitpod-overview.libsonnet') +
 (import './dashboards/gitpod-nodes-overview.libsonnet') +
 (import './dashboards/gitpod-admin-workspace.libsonnet') +
@@ -14,6 +16,5 @@
     // Example:
     // 'my-new-dashboard.json': (import 'dashboards/components/new-component.json'),
     'gitpod-cluster-autoscaler-k3s.json': (import 'dashboards/gitpod-cluster-autoscaler-k3s.json'),
-    'coredns.json': (import 'dashboards/coredns.json'),
-  },
+  } + corednsMixin.grafanaDashboards,
 }
```

File: operations/observability/mixins/cross-teams/dashboards.libsonnet
Comment (ArthurSens) [inline operations/observability/mixins/cross-teams/dashboards.libsonnet:19]: May I also suggest that we keep this dashboard in the Workspace mixin?

I believe only k3s clusters are using it at the moment

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#7826*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/observability/mixins/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/OWNERS)**
- **[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sagor999** [comment]
@ArthurSens do you know why this job fails: https://werft.gitpod-dev.com/job/gitpod-build-pavel-coredns-dashboard.14
?
It did deploy grafana (though I am having issues with trying to setup port forward into it)

**Comment by sagor999** [comment]
@ArthurSens so dashboard is working, though it doesn't have any data, since my local prometheus doesn't have any coredns to scrape. Though looking at dashboard, it will not be a great fit for us. So what I will do instead is author custom dashboard for coredns and import it into observability via json.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#7826*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ArthurSens — APPROVED** [review]
What exactly isn't a good fit?

The good thing about jsonnet is that allows us to customize stuff as needed. I can propose changes upstream if that makes sense :)

Meanwhile, no need to block this from getting merged!

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: fa073470adebe7c1b1f9c0129d29ee13adaf2713</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7966#pullrequestreview-872833416" title="Approved">ArthurSens</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **sagor999** after the PR has been reviewed.

Associated issue: *#7826*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by kylos101 — APPROVED** [review]
/LGTM

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7966#pullrequestreview-872833416" title="Approved">ArthurSens</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7966#pullrequestreview-873500589" title="LGTM">kylos101</a>*

Associated issue: *#7826*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[operations/observability/mixins/workspace/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/observability/mixins/workspace/OWNERS)~~ [kylos101]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sagor999** [comment]
@ArthurSens this is most probably because I have no experience working with jsonnet. :( So not sure how to tweak it to suit our needs.

**Pull request merged**
